src/cpu/x64/matmul_inner_product.cpp

```diff
@@ -0,0 +1,271 @@
+        { BA8a24b, {AB8b24a, AcB8b24a, AcdB8b24a, AcdeB8b24a}},
+        { BA16a16b, {AB16b16a, AcB16b16a, AcdB16b16a, AcdeB16b16a}},
+        { BA16a32b, {AB16b32a, AcB16b32a, AcdB16b32a, AcdeB16b32a}},
+        { BA16a48b, {AB16b48a, AcB16b48a, AcdB16b48a, AcdeB16b48a}},
+        { BA16a64b, {AB16b64a, AcB16b64a, AcdB16b64a, AcdeB16b64a}},
+        { BA16a16b2a, {AB16b16a2b, AcB16b16a2b, AcdB16b16a2b, AcdeB16b16a2b}},
+        { BA16a32b2a, {AB16b32a2b, AcB16b32a2b, AcdB16b32a2b, AcdeB16b32a2b}},
+        { BA16a48b2a, {AB16b48a2b, AcB16b48a2b, AcdB16b48a2b, AcdeB16b48a2b}},
+        { BA16a64b2a, {AB16b64a2b, AcB16b64a2b, AcdB16b64a2b, AcdeB16b64a2b}},
+        { BA16a16b4a, {AB16b16a4b, AcB16b16a4b, AcdB16b16a4b, AcdeB16b16a4b}},
+        { BA16a32b4a, {AB16b32a4b, AcB16b32a4b, AcdB16b32a4b, AcdeB16b32a4b}},
+        { BA16a48b4a, {AB16b48a4b, AcB16b48a4b, AcdB16b48a4b, AcdeB16b48a4b}},
+        { BA16a64b4a, {AB16b64a4b, AcB16b64a4b, AcdB16b64a4b, AcdeB16b64a4b}}};
+    // clang-format on
+
+    auto mm_wei_tag = format_tag::undef;
+    if (weights_md()->format_kind != format_kind::any) {
+        for (const auto &v : mm_wei_to_ip_wei) {
+            if (memory_desc_matches_tag(
+                        *weights_md(), v.second[weights_md()->ndims - 2])) {
+                mm_wei_tag = v.first;
+                // Check if the user-provided blocked layout can be handled.
+                const bool has_spatial = KD() + KH() + KW() > 3;
+                const int k_blk = get_k_blk(mm_wei_tag);
+                if (weights_md()->ndims > 2 && has_spatial && k_blk > 0
+                        && IC() % k_blk != 0)
+                    return status::unimplemented;
+                break;
+            }
+        }
+    } else {
+        mm_wei_tag = format_tag::any;
+    }
+    if (mm_wei_tag == format_tag::undef) return status::unimplemented;
+
+    memory_desc_t mm_src_md {};
+    memory_desc_t mm_wei_md {};
+    memory_desc_t mm_dst_md {};
+
+    if (bias_md_.format_kind == format_kind::any)
+        CHECK(memory_desc_init_by_tag(bias_md_, x));
+
+    CHECK(init_matmul_md(mm_src_md, *src_md(), format_tag::ab));
+    CHECK(init_matmul_md(mm_wei_md, *weights_md(), mm_wei_tag, true));
+    CHECK(init_matmul_md(mm_dst_md, *dst_md(), format_tag::ab));
+
+    const auto src_tag = utils::pick(src_md()->ndims - 2, ab, acb, acdb, acdeb);
+    if (src_md()->format_kind == format_kind::any)
+        CHECK(memory_desc_init_by_tag(src_md_, src_md_.ndims, src_md_.dims,
+                src_md_.data_type, src_tag));
+    else if (!memory_desc_matches_tag(*src_md(), src_tag))
+        return status::unimplemented;
+
+    const auto dst_tag = ab;
+    if (dst_md()->format_kind == format_kind::any)
+        CHECK(memory_desc_init_by_tag(dst_md_, dst_md_.ndims, dst_md_.dims,
+                dst_md_.data_type, dst_tag));
+    else if (!memory_desc_matches_tag(*dst_md(), dst_tag))
+        return status::unimplemented;
```

Comment:
You probably want all those unimplemented status returned to be wrapped in verbose call to save time figuring out which one failed.

Reply:
Done, thanks for pointing this out!